### ISSUE TITLE
Imperialism bug + 2 others

### ISSUE THREAD

**Issue by BorussiaVox** [LoneGazebo/Community-Patch-DLL#4962]
_1. Mod version (i.e Date - 4/23):_
Vox Populi (11-21-3)

_2. Mod list (if using Vox Populi only, leave blank):_
Infoaddict
reallyadvancedsetup

_3. Error description:_ 
1.  If you adopt imperialism there is no reduction of upgrade costs of units (tooltip says 10% but upgrade cost remains the same. At least the 10% unit production bonus works).

2. Naval units are offered the promotion "Navigator I" twice at the same time. 

3. Explorers have the promotion "embarkation with defense" but the Brazil UU does not have it. (makes no sense imho)

_4. Steps to reproduce (optional):_

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Comment by Vyyt02** [comment]
#1 seems to be the same as my already closed issue #4903. It seems that the fix did not work? People on CivFanatics also report that neither the Military Complex unit upgrade reduction works.

BorussiaVox, don't want to speak for Gazebo and team, but I think it is easier for them to track bugs if you create separate issues for each bug instead of packing all 3 bugs in 1 issue/report.

**Comment by LessRekkless** [comment]
Unrelated bugs go in separate issues. It helps with organization, and also helps you find bugs that have already been solved.
For example, your 2nd bug has already been reported in #4941 with the same version as you.

**Comment by BorussiaVox** [comment]
I changed the CBO/BalanceChanges/CoreDefines.sql to
'UNIT_UPGRADE_COST_DISCOUNT_MAX' - make it '-99'
 as you mentioned and tried it with my savegames (Vox Populi 11-21-3)

 Here is what I saw:
 1. If I have already adopted imperialism and all its policies, nothing changes because the games does not seem to check the values again when I load the game.

 2. When I load the old savegame and adopt imperialism it reduces the costs but only by half of the promised amount.
 Upgrading my pikeman costs 510 before adopting imperialism, after it costs 485 (5% reduction) instead of 459 (the promised 10% reduction).
 I have not checked if each policy of the imperialism tree also only reduces it by half of the promised amount (2,5% instead of 5%).

 3. When I adopt "military industrial complex" from the autocracy tree it also only reduces the upgrade costs by half of the promised amount (by 1/6 (16,5%) instead of 1/3 (33%))
 For example my Fusilier can upgrade for 700 Gold to rifleman instead of 840 -> only 16,5% reduction.
 In comparison the reduced purchasing costs of units after adopting "military industrial complex" work for the full 33% reduction --> units cost 1000 instead of 1500.
